File: app/src/main/java/com/fire/mangareader/database/AppDatabase.java
package com.fire.mangareader.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

// تم إضافة الجداول الجديدة ورفع الإصدار إلى 5
@Database(entities = {
        LibraryItem.class, 
        DownloadedChapter.class, 
        ChapterState.class, 
        CachedManga.class, 
        CachedChapter.class
}, version = 5, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {
    private static AppDatabase instance;

    public abstract MangaDao mangaDao();
    public abstract DownloadDao downloadDao();
    public abstract ChapterStateDao chapterStateDao();
    public abstract CacheDao cacheDao(); // تم إضافة الـ DAO الخاص بالتخزين الذكي

    public static synchronized AppDatabase getInstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(),
                    AppDatabase.class, "manga_fire_db")
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return instance;
    }
}

File: app/src/main/java/com/fire/mangareader/data/local/AppDatabase.kt
package com.fire.mangareader.data.local

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.fire.mangareader.data.local.entity.FavoriteManga
import com.fire.mangareader.data.local.entity.RecentManga
import com.fire.mangareader.data.local.dao.FavoriteDao
import com.fire.mangareader.data.local.dao.RecentDao

@Database(entities = [CommentEntity::class, FavoriteManga::class, RecentManga::class], version = 3, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {
    abstract fun commentDao(): CommentDao
    abstract fun favoriteDao(): FavoriteDao
    abstract fun recentDao(): RecentDao

    companion object {
        @Volatile
        private var INSTANCE: AppDatabase? = null

        fun getDatabase(context: Context): AppDatabase {
            return INSTANCE ?: synchronized(this) {
                val instance = Room.databaseBuilder(
                    context.applicationContext,
                    AppDatabase::class.java,
                    "manga_comments_database"
                )
                .fallbackToDestructiveMigration()
                .build()
                INSTANCE = instance
                instance
            }
        }
    }
}
